nuxeo-coldstorage/src/test/java/org/nuxeo/coldstorage/DummyColdStorageFeature.java

```diff
@@ -8,6 +9,6 @@
  * @since 2021.0.0
  */
 @Deploy("org.nuxeo.coldstorage.test:OSGI-INF/test-dummy-coldstorage-contrib.xml")
-@Features(ColdStorageFeature.class)
+@Features({ColdStorageFeature.class, NotificationFeature.class})
 public class DummyColdStorageFeature implements RunnerFeature {
```

Comment:
Why not add NotificationFeature in the ColdStorageFeature?

Reply:
The `NotificationFeature` is not needed in `TestS3ColdStorageService`  which uses the `ColdStorageFeature`.
I think it would be better to not add it if not needed (to avoid unforseen side effects) ?

Reply:
I would say it's not an issue to have it globally, this way, we don't need to run/introspect each test or test feature to know if it needs it. @guirenard WDYT

Reply:
OK pushing it ! If the fox catches me I'll revert it !

Reply:
It will be needed in TestS3ColdStorageService too see https://jenkins.platform.dev.nuxeo.com/job/coldstorage/job/nuxeo-coldstorage/job/lts-2023/24/consoleFull
```
00:10:27.783  [INFO] Running org.nuxeo.coldstorage.service.TestS3ColdStorageService
00:10:28.357  2024-01-29 10:43:35,642 [main] ERROR [ComponentManagerImpl] Duplicate component name: service:org.nuxeo.ecm.core.blob.BlobManager
00:10:28.357  RuntimeStreamFeature: Deploying Nuxeo Stream using Mem
00:10:28.357  WorkManagerFeature: Deploying WorkManager using default implementation
00:10:28.357  StorageConfiguration: Deploying JDBC using DatabaseH2
00:10:28.357  StorageConfiguration: Deploying a DBS repository using mongodb
00:10:29.745  2024-01-29 10:43:36,995 [main] ERROR [AutomationComponent] Operation chain with id 'xmlExportRendition' references unknown operation with id 'Document.Export'
00:10:29.745  2024-01-29 10:43:36,996 [main] ERROR [AutomationComponent] Operation chain with id 'zipTreeExportRendition' references unknown operation with id 'Document.Export'
00:10:29.745  2024-01-29 10:43:37,005 [main] WARN  [NotificationService] Your GeneralSettingsDescriptor has been contributed with a custom mailSessionJndiName. This field is now deprecated. Please use mailSenderName and contribute a MailSenderDescriptor
00:10:29.745  2024-01-29 10:43:37,005 [main] ERROR [RegistrationInfoImpl] Component service:org.nuxeo.ecm.platform.ec.notification.service.NotificationService notification of application started failed: Cannot invoke "org.nuxeo.mail.MailServiceImpl.registerJndiSMTPSender(String)" because the return value of "org.nuxeo.runtime.api.Framework.getService(java.lang.Class)" is null
00:10:29.745  java.lang.NullPointerException: Cannot invoke "org.nuxeo.mail.MailServiceImpl.registerJndiSMTPSender(String)" because the return value of "org.nuxeo.runtime.api.Framework.getService(java.lang.Class)" is null
```